h2o-algos/src/main/java/hex/rulefit/RuleFit.java

```diff
@@ -307,14 +307,53 @@ SharedTreeModel trainTreeModel(RuleFitModel.Algorithm algorithm, int maxDepth) {
            if (deviance.length < 5) {
                bestLambdaIndex = deviance.length - 1;
            } else {
-               bestLambdaIndex = getBestLambdaIndex(deviance, lambdas);
+               bestLambdaIndex = getBestLambdaIndex(deviance);
+               if (bestLambdaIndex >= lambdas.length) {
+                   bestLambdaIndex = getBestLambdaIndexCornerCase(deviance, lambdas);
+               }
            }
 
             lambdaModel.remove();
             return new double[]{lambdas[bestLambdaIndex]};
         }
 
-        int getBestLambdaIndex(double[] deviance, double[] lambdas) {
+        int getBestLambdaIndex(double[] deviance) {
+            int bestLambdaIndex = deviance.length - 1;
+            if (deviance.length >= 5) {
+                double[] array = differenceArray(signArray(differenceArray(differenceArray(deviance))));
+                for (int i = 0; i < array.length; i++) {
+                    if (array[i] != 0 && i > 0) {
+                        bestLambdaIndex = 3 * i;
```

Comment:
Hungarian constant a.k.a Pištéty's constant found cc @honzasterba . This time of value 3 ! :rofl:

Reply:
this is in line with https://github.com/h2oai/h2o-tutorials/blob/8df6b492afa172095e2595922f0b67f8d715d1e0/best-practices/explainable-models/rulefit.py. I can imagine something clearer, but since I haven't found any better working method, I would stick to this.

Reply:
Sure. Just a little bit of fun, as I had no idea where this comes from.